build_tools/python/e2e_test_framework/definitions/utils.py

```diff
@@ -0,0 +1,50 @@
+## Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Utils that help construct definitions."""
+
+from typing import Callable, List, Sequence
+
+MAX_SUBSTITUTION_ITERATIONS = 10
+
+
+def transform_flags(flags: Sequence[str],
+                    map_funcs: Sequence[Callable[[str], str]]) -> List[str]:
+  """Call map functions to transform flag values, e.g., replace placeholders
+  that were unknown when the flag was constructed.
+
+  It parses and extracts the flag values from both keyword and positional flags,
+  transforms them, and returns the updated flags with transformed values.
+
+  Each flag value is transformed only once by each map function in order.
+  
+  Args:
+    flags: list of flags.
+    map_funcs: list of map functions to map flag value.
+  Returns:
+    List of transformed flags.
+  """
+
+  transformed_flags = []
+  for flag in flags:
```

Comment:
Should we use argparse instead of doing our own parsing here?

Reply:
Not very sure argparse will be helpful here. argparse needs to specify the valid arguments, and it does special handling on things like `--`